poller-lambdas/src/aws.ts

```diff
@@ -0,0 +1,44 @@
+import { SecretsManagerClient } from '@aws-sdk/client-secrets-manager';
+import { SendMessageCommand, SQSClient } from '@aws-sdk/client-sqs';
+import { POLLER_LAMBDA_ENV_VAR_KEYS } from '../../shared/pollers';
+import {
+	getEnvironmentVariableOrCrash,
+	isRunningLocally,
+	remoteAwsConfig,
+} from './config';
+
+const buildLocalFakeSqsClient = () => {
+	const queueData: Record<string, SendMessageCommand[]> = {};
+	return {
+		queueData: queueData,
+		send: (command: SendMessageCommand) => {
+			const currentQueue: SendMessageCommand[] =
+				queueData[command.input.QueueUrl!] ?? [];
+			queueData[command.input.QueueUrl!] = [...currentQueue, command];
+			return Promise.resolve();
+		},
+	};
+};
+
+export const sqs = isRunningLocally
+	? buildLocalFakeSqsClient()
+	: new SQSClient(remoteAwsConfig);
+
+export const secretsManager = new SecretsManagerClient(remoteAwsConfig);
+
+const lambdaApp = process.env['App'];
+const ownQueueUrl = getEnvironmentVariableOrCrash(
+	POLLER_LAMBDA_ENV_VAR_KEYS.OWN_QUEUE_URL,
+);
+
+export const queueNextInvocation = (props: {
+	MessageBody: string;
+	DelaySeconds?: number;
+}) =>
+	sqs.send(
+		new SendMessageCommand({
+			QueueUrl: ownQueueUrl,
```

Comment:
when testing locally, `ownQueueUrl` here is false, and I believe that this is because it's set when this file is imported by `localRun`, which is before the env var has been set, within the main run function of `localRun`.

if we move line 30 to inside the `queueNextInvocation` block then it will get the right value when run locally

Reply:
many thanks Pete, I've addressed this amongst other things in https://github.com/guardian/newswires/pull/83/commits/841e6f0fc31dcf920a7e416a9e33e5219464afe0